### ISSUE TITLE
TextInput ignores dynamic textAlign with secureTextEntry

### ISSUE THREAD

**Issue by pzyryanov1995** [facebook/react-native#23059]
- [x] Review the documentation: https://facebook.github.io/react-native
  - [x] Search for existing issues: https://github.com/facebook/react-native/issues
  - [ ] Use the latest React Native release: https://github.com/facebook/react-native/releases

## Environment
React Native Environment Info:
    System:
      OS: macOS Sierra 10.12.6
      CPU: x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
      Memory: 108.94 MB / 8.00 GB
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 10.14.2 - /usr/local/bin/node
      npm: 6.5.0 - /usr/local/bin/npm
    SDKs:
      iOS SDK:
        Platforms: iOS 11.2, macOS 10.13, tvOS 11.2, watchOS 4.2
    IDEs:
      Xcode: 9.2/9C40b - /usr/bin/xcodebuild
    npmPackages:
      react: 16.5.0 => 16.5.0
      react-native: 0.57.1 => 0.57.1
    npmGlobalPackages:
      react-native-git-upgrade: 0.2.7

## Description
Reproduce only for ios TextInput with secureTextEntry.

This problem is like this: [https://github.com/facebook/react-native/issues/12160](similiar_issue).
On focus, text align should be in center with highlight border in bottom.
On blur, text align should be in left position. But the text remains in center.

For android it's work perfectly.

If add key like this: 

`key={`shouldCenter${this.state.focused}`}`

focus on input work only after second press, see in [Demo](https://snack.expo.io/H163nTCzV)

## Reproducible Demo
https://snack.expo.io/H1ZOFpRGN

```
import * as React from 'react';
import { Text, View, StyleSheet, TextInput } from 'react-native';

export default class App extends React.Component {
  constructor(props) {
      super(props);
      this.state = {
          focused: false,
          text: '',
      };
  }

  render() {
    const shouldCenter = this.state.focused;
    return (
      <View style={styles.container}>
        <Text>{`center: ${shouldCenter}`}</Text>
        <TextInput
          value={this.state.text}
          style={[
            styles.inputDefault,
            shouldCenter
              ? styles.inputFocused
              : styles.inputBlured
          ]}
          onFocus={() => this.setState({ focused: true })}
          onBlur={() => this.setState({ focused: false })}
          onChangeText={text => this.setState({ text })}
          secureTextEntry
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      backgroundColor: '#ecf0f1',
      padding: 8,
    },
    inputDefault: {
        color: '#000',
        fontSize: 16,
        borderBottomWidth: 1,
        width: '100%',
    },
    inputFocused: {
        textAlign: 'center',
        borderBottomColor: '#F86C30',
    },
    inputBlured: {
        textAlign: 'left',
        borderBottomColor: '#a3a3a3',
    },
});
```

**Comment by pzyryanov1995** [comment]
I found a temporary solution, but it is incorrect.
In textInput added this code:
```

          onFocus={() => {
            setTimeout(() => {
                this.input.focus();
            }, 0);
            this.setState({ focused: true });
          }}
          key={`shouldCenter${shouldCenter}`}
          ref={(ref) => { this.input = ref; }}
```
Result demo: [https://snack.expo.io/B174vzk74](result_code)

**Comment by ericlewis** [comment]
I think this may be fixed by: #23524
